src/pyproject_hooks/_impl.py

```diff
@@ -23,8 +23,12 @@ def read_json(path):
 class BackendUnavailable(Exception):
     """Will be raised if the backend cannot be imported in the hook process."""
 
-    def __init__(self, traceback):
+    def __init__(self, traceback, message=None, backend_name=None, backend_path=None):
+        # Preserving arg order for the sake of API backward compatibility.
+        self.backend_name = backend_name
+        self.backend_path = backend_path
         self.traceback = traceback
+        super().__init__(message or "Error while importing backend")
 
 
 class BackendInvalid(Exception):
```

Comment:
Thank you very much for the review @pradyunsg.

I was considering that removing `BackendInvalid` would cause backward compatibility problems (e.g. it seems to be in `__init__.py`'s `__all__`, some folks might consider that public API, so it might break someone's script if they do a `from pyproject_hooks import BackendInvalid`).

I am happy to remove the class if the project is happy to take the (potential) backwards incompatibility. Alternative, also happy to change the docstring to "deprecated" or something like that. What is your take on that?

Reply:
TBH, I don't [see anyone using this on GitHub](https://github.com/search?utf8=%E2%9C%93&q=%2Fimport+.%2B%28%5Cn.%2B%29%3F+BackendInvalid%2F+language%3APython&type=code) so it seems reasonably safe to remove.

We can bump the major version since this is backwards-incompatible (and version numbers are cheap), and then we don't have to bother with having dead code in the codebase that no one relies on.

Reply:
> it seems to be in `__init__.py`'s `__all__`, some folks might consider that public API,

FWIW, I'd be surprised if anyone doesn't consider it public API. 😅

Reply:
No problems, in a318f8ed39b9f01d4c81066bbf4966c377a613b2, I removed the unused exception class.

In the case a deprecation cycle is preferred we can just revert this commit.